theseus/geometry/se3.py

```diff
@@ -176,14 +174,23 @@ def _hat_matrix_check(matrix: torch.Tensor):
         if matrix.ndim != 3 or matrix.shape[1:] != (4, 4):
             raise ValueError("Hat matrices of SE3 can only be 4x4 matrices")
 
-        if matrix[:, 3].abs().max().item() > HAT_EPS:
-            raise ValueError("The last row of hat matrices of SE3 can only be zero.")
-
-        if (
-            matrix[:, :3, :3].transpose(1, 2) + matrix[:, :3, :3]
-        ).abs().max().item() > HAT_EPS:
-            raise ValueError(
-                "The 3x3 top-left corner of hat matrices of SE3 can only be skew-symmetric."
+        if _LieGroupCheckContext.get_context():
+            if matrix[:, 3].abs().max().item() > HAT_EPS:
+                raise ValueError(
+                    "The last row of hat matrices of SE3 can only be zero."
+                )
+
+            if (
+                matrix[:, :3, :3].transpose(1, 2) + matrix[:, :3, :3]
+            ).abs().max().item() > HAT_EPS:
+                raise ValueError(
+                    "The 3x3 top-left corner of hat matrices of SE3 can only be skew-symmetric."
+                )
+        else:
+            warnings.warn(
+                "functorch is enabled and the skew-symmetry of hat matrices is "
```

Comment:
Need to change this message as well. How about "Lie group checks are disabled, so the skew-symmetry... etc."?